contracts/LiquidityBridgeContract.sol

```diff
@@ -193,15 +197,30 @@ contract LiquidityBridgeContract {
         @param quote The quote that identifies the service
         @return Boolean indicating whether the call was successful
      */
+	 // SDL I don't like this method to be able to be called recursively
+	 // even if I can't see a vuln in the  logic. Defensive programming should
+	 // be applied.
     function callForUser(Quote memory quote) external payable onlyRegistered returns (bool) {
         require(msg.sender == quote.liquidityProviderRskAddress, "Unauthorized");
         require(balances[quote.liquidityProviderRskAddress] + msg.value >= quote.value, "Insufficient funds");   
         require(address(this) == quote.lbcAddress, "Wrong LBC address");  
-        require(collateral[msg.sender] >= minCol, "Insufficient collateral");   
-
+        require(collateral[msg.sender] >= minCol, "Insufficient collateral"); // SDL: Why this check? If the user accepted, it's already too late. Better if the LP calls callForUser() than if he is forbitten to.  
+		
+		// SDL: The subtraction may underflow (both args are uint). 
+		// I think this version
+		// of Solidity will revert on underflow.
+		// Do we want this to occur?
+		// I think we want to allow msg.value to be zero, and to consume
+		// the pre-existent balance. In that case then this operation 
+		// should be performed in ints instead of uints (unless Solidity is
+		// already casting the subtraction to int, but I don't think so)
+		// Is there any test case for this ?
         balances[quote.liquidityProviderRskAddress] += msg.value - quote.value;
-        emit BalanceIncrease(quote.liquidityProviderRskAddress, msg.value);
+        
+		// SDL: Is this event really needed? What for ?
+		emit BalanceIncrease(quote.liquidityProviderRskAddress, msg.value);
 
+		// SDL: Shoudn't here by a "gap" added to gasLimit ?
```

Comment:
I mean, shouldn't this be something like:

require(gasleft() +someExtraGas>= quote.gasLimit, "Insufficient gas");

Reply:
I am not sure, what's the point of the extra gas?

You suggested me to add this check not to ensure that the transaction has enough gas to finish execution, but to guarantee that the LP cannot perform this call with less gas than the agreed amount. This is because RSK will not raise an error when performing a call with less gas left than the defined gas limit.

Reply:
I wrote it wrong. I meant

require(gasleft() >= quote.gasLimit +someExtraGas, "Insufficient gas");